src/v/utils/moving_average.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+#include <array>
+#include <numeric>
+
+/*
+ * simple moving average with configurable sample count
+ */
+
+template<typename T, size_t Samples>
+class moving_average {
+public:
+    explicit moving_average(T initial)
+      : _value(initial) {
+        for (auto& s : _samples) {
+            s = _value;
+        }
+    }
+
+    void update(T v) {
+        _samples[_idx] = v;
+        _idx = (_idx + 1) % Samples;
+        _value = std::accumulate(_samples.begin(), _samples.end(), T{0})
```

Comment:
This is skewed until it's full. You could use `boost::static_vector` and `.size()`.

Reply:
great idea